pkg/util/objectstore/store.go

```diff
@@ -335,3 +336,17 @@ func (s *ObjectStore[T]) processDeleteQueueItem(objectDeletionKey *objectKey) bo
 func (s *ObjectStore[T]) HasSynced() bool {
 	return s.hasSynced()
 }
+
+// WaitForStoreSync waits for stores to sync. It returns an error if the context is cancelled. You
+// need to provide the HasSynced method for each store you want to wait on.
+func WaitForStoreSync(ctx context.Context, storeSyncs ...func() bool) error {
```

Comment:
probably better to name it WaitForStoreSyncs?

Reply:
Done